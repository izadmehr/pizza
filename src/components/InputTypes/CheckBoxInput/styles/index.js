import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  align-items: center;
`;
export const TextInput = styled.input`
  border: none;
  font-size: 1rem;
  padding: 0 5px;
  background-color: transparent;
  caret-color: ${props => props.theme.color.primary};

  ::-ms-clear {
    display: none;
  }

  ::-webkit-input-placeholder {
    font-size: 1rem;
    font-weight: 500;
    color: #7f7f7f;
    font-family: Barlow, sans-serif;
  }

  ::placeholder {
    font-size: 1rem;
    font-weight: 500;
    color: #7f7f7f;
    font-family: Barlow, sans-serif;
  }
  :focus {
    outline: none;
  }
`;
