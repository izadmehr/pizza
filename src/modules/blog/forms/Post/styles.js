import styled from "styled-components";

import { H3 } from "../../../../components/Typography";

export const PizzaSizeSelector = styled.div`
  padding: 0 1rem;
  display: flex;
  position: relative;
  align-items: center;
  height: 2.5rem;
  border: solid 1px ${props => props.theme.color.lightGrey};
  :focus-within {
    border: solid 1px ${props => props.theme.color.primary};
  }
  width: 400px;
`;

export const InputContainer = styled.div`
  flex: 1;
`;
export const ResultsContainer = styled.ul`
  position: absolute;
  margin-top: 0.4rem;
  box-shadow: -1px 0 10px 0 rgba(0, 0, 0, 0.1), 0 0 10px 0 rgba(0, 0, 0, 0.1);
  padding: 0;
  background: white;
  list-style-type: none;
`;

export const ResultItem = styled.li`
  padding: 1rem 2rem;
  background: ${({ highlighted }) =>
    highlighted ? "rgba(0,0,0,.03)" : "white"};
  font-weight: ${({ selected }) => (selected ? "bold" : "normal")};
`;

export const PizzaSizesContainer = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const Toppings = styled(H3)`
  margin-top: 1rem;
`;

export const TappingContainer = styled.div`
  margin-top: 0.5rem;
`;
