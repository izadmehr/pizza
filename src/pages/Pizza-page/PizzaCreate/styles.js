import styled from "styled-components";

export const PizzaSizeSelector = styled.div`
  padding: 0 1rem;
  display: flex;
  position: relative;
  align-items: center;
  height: 2.5rem;
  border: solid 1px ${props => props.theme.color.lightGrey};
  :focus-within {
    border: solid 1px ${props => props.theme.color.primary};
  }
  width: 400px;
`;

export const InputContainer = styled.div`
  flex: 1;
`;
export const ResultsContainer = styled.ul`
  position: absolute;
  margin-top: 0.4rem;
  box-shadow: -1px 0 10px 0 rgba(0, 0, 0, 0.1), 0 0 10px 0 rgba(0, 0, 0, 0.1);
  padding: 0;
  background: white;
  list-style-type: none;
`;

export const ResultItem = styled.li`
  padding: 1rem 2rem;
  background: ${({ highlighted }) =>
    highlighted ? "rgba(0,0,0,.03)" : "white"};
  font-weight: ${({ selected }) => (selected ? "bold" : "normal")};
`;
