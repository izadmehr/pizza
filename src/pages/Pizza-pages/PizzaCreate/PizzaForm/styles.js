import styled from "styled-components";

import { H3 } from "components/Typography";

export const PizzaSizesContainer = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const Toppings = styled(H3)`
  margin-top: 1rem;
`;

export const TappingContainer = styled.div`
  margin-top: 0.5rem;
`;
