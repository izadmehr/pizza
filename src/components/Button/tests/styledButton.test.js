import React from "react";
import { mount } from "enzyme";

import { theme } from "styles";

import "jest-styled-components";
import "jsdom-global/register";
import { StyledButton } from "../index";

describe("StyledButton Component", () => {
  it("has correct default styles", () => {
    const wrapper = mount(
      <StyledButton theme={theme}>Button Text</StyledButton>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("has correct :active background-color for bgType", () => {
    const wrapper = mount(
      <StyledButton bgType="success" theme={theme}>
        Button Text
      </StyledButton>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
