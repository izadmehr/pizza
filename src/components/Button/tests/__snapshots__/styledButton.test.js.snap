// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledButton Component has correct :active background-color for bgType 1`] = `
.c0 {
  background-color: #48cfad;
  border: none;
  color: white;
  padding: 12px 20px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  font-size: 14px;
  cursor: pointer;
  -webkit-transition: background-color 150ms linear;
  transition: background-color 150ms linear;
  margin-right: 2px;
}

.c0:hover {
  background-color: #37bc9b;
}

.c0:focus {
  outline: 0;
}

.c0:active {
  background-color: #ccd1d9;
}

<styled.button
  bgType="success"
  theme={
    Object {
      "color": Object {
        "black": "#000",
        "darkGreen": "#37bc9b",
        "darkGrey": "#aab2bd",
        "darkRed": "#da4453",
        "lightGreen": "#48cfad",
        "lightGrey": "#ccd1d9",
        "lightRed": "#ed5565",
        "primary": "#5d9cec",
        "secondary": "#4a89dc",
        "white": "#fff",
      },
      "layout": Object {
        "maxWidth": "62.5rem",
        "minWidth": "25rem",
      },
    }
  }
>
  <StyledComponent
    bgType="success"
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  background-color: ",
            [Function],
            ";
  border: none;
  color: white;
  padding: 12px 20px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 14px;
  cursor: pointer;
  transition: background-color 150ms linear;
  margin-right: 2px;

  &:hover {
    background-color: ",
            [Function],
            ";
  }

  &:focus {
    outline: 0;
  }

  &:active {
    background-color: ",
            [Function],
            ";
  }
",
          ],
        },
        "displayName": "styled.button",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "button",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    theme={
      Object {
        "color": Object {
          "black": "#000",
          "darkGreen": "#37bc9b",
          "darkGrey": "#aab2bd",
          "darkRed": "#da4453",
          "lightGreen": "#48cfad",
          "lightGrey": "#ccd1d9",
          "lightRed": "#ed5565",
          "primary": "#5d9cec",
          "secondary": "#4a89dc",
          "white": "#fff",
        },
        "layout": Object {
          "maxWidth": "62.5rem",
          "minWidth": "25rem",
        },
      }
    }
  >
    <button
      className="c0"
    >
      Button Text
    </button>
  </StyledComponent>
</styled.button>
`;

exports[`StyledButton Component has correct default styles 1`] = `
.c0 {
  background-color: #5d9cec;
  border: none;
  color: white;
  padding: 12px 20px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  font-size: 14px;
  cursor: pointer;
  -webkit-transition: background-color 150ms linear;
  transition: background-color 150ms linear;
  margin-right: 2px;
}

.c0:hover {
  background-color: #4a89dc;
}

.c0:focus {
  outline: 0;
}

.c0:active {
  background-color: #5d9cec;
}

<styled.button
  theme={
    Object {
      "color": Object {
        "black": "#000",
        "darkGreen": "#37bc9b",
        "darkGrey": "#aab2bd",
        "darkRed": "#da4453",
        "lightGreen": "#48cfad",
        "lightGrey": "#ccd1d9",
        "lightRed": "#ed5565",
        "primary": "#5d9cec",
        "secondary": "#4a89dc",
        "white": "#fff",
      },
      "layout": Object {
        "maxWidth": "62.5rem",
        "minWidth": "25rem",
      },
    }
  }
>
  <StyledComponent
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  background-color: ",
            [Function],
            ";
  border: none;
  color: white;
  padding: 12px 20px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 14px;
  cursor: pointer;
  transition: background-color 150ms linear;
  margin-right: 2px;

  &:hover {
    background-color: ",
            [Function],
            ";
  }

  &:focus {
    outline: 0;
  }

  &:active {
    background-color: ",
            [Function],
            ";
  }
",
          ],
        },
        "displayName": "styled.button",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "button",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    theme={
      Object {
        "color": Object {
          "black": "#000",
          "darkGreen": "#37bc9b",
          "darkGrey": "#aab2bd",
          "darkRed": "#da4453",
          "lightGreen": "#48cfad",
          "lightGrey": "#ccd1d9",
          "lightRed": "#ed5565",
          "primary": "#5d9cec",
          "secondary": "#4a89dc",
          "white": "#fff",
        },
        "layout": Object {
          "maxWidth": "62.5rem",
          "minWidth": "25rem",
        },
      }
    }
  >
    <button
      className="c0"
    >
      Button Text
    </button>
  </StyledComponent>
</styled.button>
`;
